### PR TITLE
implement retrying for global adapter connection

### LINKED ISSUE #2143: Implement Retries for Adapter - global Adapter XDS communication
### Describe your problem(s)
If the connection is dropped or the global adapter restarts, the local adapter should be able to reconnect and continue the process.

### How will you implement it

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
If the connection fails at start or in the middle, the adapter will retry with a linear backoff.

regarding that xds retry mechanism for LA-GA,

1.  If the network remains disconnected (using docker network disconnect command) then after a ~5min period, the connection is broken and it would be retrying. (this one is getting recovered)
2.  If we disconnect adapter from network, do an update within ga, wait for sometime
2.i)  no xds update is received. if it continues, it would end up with 1 scenario.
2.ii)  if we connect the adapter again, then the xds update is recieved immediately after connect

### Issues

Fixes #2143
related to #2106

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/ga/client.go [hunk 1]
```diff
@@ -219,3 +226,39 @@ func consumeAPIChannel() {
...
+	conf, _ := config.ReadConfigs()
+	retryIntervalSeconds := conf.GlobalAdapter.RetryInterval
+	backOff := grpc_retry.BackoffLinearWithJitter(retryIntervalSeconds*time.Second, 0.5)
+	logger.LoggerGA.Infof("Dialing Global Adapter GRPC Service : %s", conf.GlobalAdapter.ServiceURL)
+	return grpc.Dial(
+		conf.GlobalAdapter.ServiceURL,
+		grpc.WithInsecure(),
+		grpc.WithBlock(),
+		grpc.WithStreamInterceptor(
+			grpc_retry.StreamClientInterceptor(grpc_retry.WithBackoff(backOff))))
```
Comment (praminda) [inline adapter/internal/ga/client.go:263]:
```suggestion
		grpc.WithStreamInterceptor(
			grpc_retry.StreamClientInterceptor(grpc_retry.WithBackoff(backOff))
		)
	)
```

**Comment by VirajSalaka** [comment]
related to wso2-enterprise/choreo#4332

**Comment by VirajSalaka** [comment]
If the network is switched (using docker network command), the connection is not immediately dropped. Rather somewhere in between 3-5 min the client itself disconnects. Then the retrying would happen until the global adapter node becomes available.

File: adapter/internal/ga/client.go (hunk 1 above)
Comment (VirajSalaka) [reply adapter/internal/ga/client.go:263]: changing parentheses ends up with adapter compilation error

**Review by praminda — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20210825.10

**Comment by choreo-cicd** [comment]
[succeeded] prod-deployment : 20210831.1

**Pull request merged**